backend/src/rss.rs

```diff
@@ -0,0 +1,243 @@
+    let video_items = match generate_video_items(
+        &db, &series_oc_id, &series_title, &rss_link, &tobira_url).await {
+        Ok(items) => items,
+        Err(_) => return Err(bad_request("empty series")),
+    };
+
+    for item in video_items {
+        // I have no idea how to correctly handle items that are resulting in an error.
+        // Pretty sure however that this here is a bad solution and would result in
+        // potentially broken feeds.
+        let _ = channel.add_child(item);
+    }
+
+    rss.add_child(channel).unwrap();
+    xml.set_root_element(rss);
+    let mut writer: Vec<u8> = Vec::new();
+    xml.generate(&mut writer).unwrap();
+
+    let xml_string = String::from_utf8(writer).unwrap();
+    
+    Ok(xml_string)
+}
+
+/// Generates the single video items of a series in Tobira for inclusion in an RSS feed.
+async fn generate_video_items(
+    db: &Client,
+    series_oc_id: &str,
+    series_title: &str,
+    rss_link: &str,
+    tobira_url: &HttpHost,
+) -> Result<Vec<XMLElement>, Error> {
+    let selection = Event::select();
+    let query = format!("select {selection} from events where part_of = $1");
+    let rows = db.query_raw(&query, dbargs![&series_oc_id]).await?;
+
+    let mut video_items: Vec<XMLElement> = Vec::new();
+
+    rows.try_for_each(|row| {
+        let event = Event::from_row_start(&row);
+        
+        let mut buf = [0; 11];
+        let tobira_event_id = event.id.to_base64(&mut buf);
+        let event_link = format!("{}/!v/{}", tobira_url, tobira_event_id);
+
+        let preferred_track = choose_preferred_track(event.tracks).unwrap();
+        let enclosure_url = &preferred_track.uri;
+        let mimetype = &preferred_track.mimetype.unwrap();
+        let thumbnail = &event.thumbnail_url.unwrap_or_default();
+
+        let mut item = XMLElement::new("item");
+
+        let item_data = [
+            ("title", vec![("", event.title)]),
+            ("link", vec![("", event_link.clone())]),
+            ("description", vec![("", event.description.unwrap_or_default())]),
+            ("dc:creator", vec![("", event.creators.join(", "))]),
+            ("pubDate", vec![("", event.created.to_rfc2822())]),
+            ("guid", vec![("", event_link)]),
+            ("media:thumbnail", vec![("url", thumbnail.to_string())]),
+            ("itunes:image", vec![("href", thumbnail.to_string())]),
+            ("enclosure", vec![
+                ("url", enclosure_url.to_string()),
+                ("type", mimetype.to_string()),
+                ("length", "0".to_string()),
+            ]),
+            ("source", vec![
+                ("url", rss_link.to_string()),
+                ("", series_title.to_string()),
+            ]),
+            ("media:content", vec![
+                ("url", enclosure_url.to_string()),
+                ("type", mimetype.to_string()),
+            ]),
+        ];
+        add_elements_to_xml(&item_data, &mut item);
+
+        video_items.push(item);
```

Comment:
Now that you are pushing to a `Vec` instead of appending to a `String`, you want to probably replace `for_each` with `map_ok` and then `try_collect`.